### PR TITLE
breaking change: remove deprecated Action items and update docs

### PR DESCRIPTION
**Pull request by drc38**
### Changes included in this PR 

***Breaking change***
As per the warning in 1.0.0 this PR removes deprecated Action items from the `enums.py` files to coincide with the 2.0.0 release, updates code references and updates documentation to be consistent with the change including string references.

For example:
 - replaces deprecated Action enums such as `Action.BootNotification` with `Action.boot_notification`
 - replaces string references such as `@on("BootNotification")` and `action="BootNotification"` with `Action.boot_notification`

If users of the library have not already migrated to the new Action enums they will need to do so before using 2.0.0.

### Checklist

1. [x] Does your submission pass the existing tests?
2. [ ] Are there new tests that cover these additions/changes? 
3. [ ] Have you linted your code locally before submission?

### CODE REVIEW

File: ocpp/charge_point.py [hunk 1]
```diff
@@ -170,20 +170,22 @@ def _raise_key_error(action, version):
     from ocpp.v201.enums import Action as v201_Action
 
     if version == "1.6":
-        if hasattr(v16_Action, action):
+        try:
```
Comment (jainmohit2001) [inline ocpp/charge_point.py:173]: Any particular reason why we are changing this piece of code? I think the previous version will still work.
Will the change bring any additional functionality?

File: ocpp/charge_point.py [hunk 2]
```diff
@@ -170,20 +170,22 @@ def _raise_key_error(action, version):
...
+            v16_Action(action)
             raise NotImplementedError(
```
Comment (ajmirsky) [inline ocpp/charge_point.py:175]: using try/except to determine if an error should be raised (the purpose of the function `_raise_key_error`) doesn't seem the best approach. also, try/except is used to catch _unexpected_ expected exceptions within other code wrapped inside, raising an explicit exception within the block isn't great practice.

**Review by ajmirsky — CHANGES_REQUESTED** [review]
given the name (and purpose) of the function `_raise_key_error`, standard checking using if/else makes more sense. if using try/except blocks, then we should just remove this function and add this try/except to the calling code.

File: ocpp/charge_point.py (hunk 2 above)
Comment (drc38) [reply ocpp/charge_point.py:175]: Note the try/except blocks are being used to catch expected _exceptions_.

Comment (jainmohit2001) [reply ocpp/charge_point.py:175]: I don't think we have any other option that to check the membership using try/except clauses here.

File: ocpp/charge_point.py (hunk 1 above)
Comment (jainmohit2001) [reply ocpp/charge_point.py:173]: @drc38, can we add a test in `test_charge_point.py` for the `_raise_key_error` function?

File: ocpp/charge_point.py (hunk 2 above)
Comment (jainmohit2001) [reply ocpp/charge_point.py:175]: Resolving this. We will stick with try/except clause for now.

**Comment by jainmohit2001** [comment]
Need to update the following:
- `docs/source/central_system.rst`. Replace `Action.BootNotification` with `Action.boot_notification`.
- `examples/v16/central_system.py`. Replace `Action.BootNotification` with `Action.boot_notification`.
- Update all the comments mentioned in `ocpp/routing.py` where `Action.` keyword is present.
- `examples/v201/central_system.py`: Replace `@on("BootNotification")` and `@on("Heartbeat")` with respective Action enums.
- `tests/test_exceptions.py`: Replace `action="BootNotification"` with relevant Action enum.
- `tests/v201/conftest.py`: Replace `action="BootNotification"` and `action="Heartbeat"` with relevant Action enums.
- `tests/test_messages.py`: Replace `action="Heartbeat"` with relevant Action enum.

File: ocpp/charge_point.py (hunk 1 above)
Comment (drc38) [reply ocpp/charge_point.py:173]: > @drc38, can we add a test in `test_charge_point.py` for the `_raise_key_error` function?

This is already tested for, see https://github.com/mobilityhouse/ocpp/commit/f159cf7b664339a73c6733c35f43a9c38b558561

**Comment by drc38** [comment]
> Need to update the following:
> 
> * `docs/source/central_system.rst`. Replace `Action.BootNotification` with `Action.boot_notification`.
> * `examples/v16/central_system.py`. Replace `Action.BootNotification` with `Action.boot_notification`.
> * Update all the comments and docs mentioned in `ocpp/routing.py` where `Action.` keyword is present.
> * `examples/v201/central_system.py`: Replace `@on("BootNotification")` and `@on("Heartbeat")` with respective Action enums.
> * `tests/test_exceptions.py`: Replace `action="BootNotification"` with relevant Action enum.
> * `tests/v201/conftest.py`: Replace `action="BootNotification"` and `action="Heartbeat"` with relevant Action enums.
> * `tests/test_messages.py`: Replace `action="Heartbeat"` with relevant Action enum.
> 
> Let's create another PR for the doc changes. For the rest, let's update them in this PR itself.

Just noticed your comment on a separate PR for doc changes, I've included them all in the below commit. Hopefully that's ok as then when this is merged all Action references are consistent.

**Comment by jainmohit2001** [comment]
> > Need to update the following:
> > 
> > * `docs/source/central_system.rst`. Replace `Action.BootNotification` with `Action.boot_notification`.
> > * `examples/v16/central_system.py`. Replace `Action.BootNotification` with `Action.boot_notification`.
> > * Update all the comments and docs mentioned in `ocpp/routing.py` where `Action.` keyword is present.
> > * `examples/v201/central_system.py`: Replace `@on("BootNotification")` and `@on("Heartbeat")` with respective Action enums.
> > * `tests/test_exceptions.py`: Replace `action="BootNotification"` with relevant Action enum.
> > * `tests/v201/conftest.py`: Replace `action="BootNotification"` and `action="Heartbeat"` with relevant Action enums.
> > * `tests/test_messages.py`: Replace `action="Heartbeat"` with relevant Action enum.
> > 
> > Let's create another PR for the doc changes. For the rest, let's update them in this PR itself.
> 
> Just noticed your comment on a separate PR for doc changes, I've included them all in the above commit. Hopefully that's ok as then when this is merged all Action references are consistent.

No worries. We can go with the doc changes in this PR itself.

File: ocpp/charge_point.py (hunk 1 above)
Comment (jainmohit2001) [reply ocpp/charge_point.py:173]: Ahh. Missed that. Looks good then.

**Comment by jainmohit2001** [comment]
LGTM. @drc38, can you please update the description for the PR and include relevant changes done in this PR.
Also, please include the fact that this is a breaking change and the impact that it will have.
Thanks!

**Comment by drc38** [comment]
> HI @drc38, can you please rebase this with the master and update datatypes/enums used (if any).

Master has been merged and tests are still passing.

**Review by jainmohit2001 — APPROVED** [review]

**Pull request merged**